include/core/shape.h

```diff
@@ -133,6 +134,950 @@ class Shape : public AutoDerivativeFunction<dim>
     const typename DoFHandler<dim>::active_cell_iterator cell,
     const unsigned int                                   component = 0);
 
+  /**
+   * @brief Return the smoothed maximum of two variables used for shape contact calculation.
+   * @param a first variable
+   * @param b second variable
+   */
+  double
+  smooth_max(double a, double b)
+  {
+    double smooth_factor = 10;
```

Comment:
seems like a magic number,. Maybe you could have the smooth max take an optional thirdargument with a default value of 10 ? This way you would get the best of both world :).